Modules/Typescript/algo/first/priorQueueFactory.ts

```diff
@@ -0,0 +1,75 @@
+interface IQueueElement {
+  element: any,
+  priority: number
+}
+
+export function PriorityQueueFactory() {
+  const items: IQueueElement[] = [];
+
+  return {
+    enqueue,
+    dequeue,
+    front,
+    isEmpty,
+    size,
+    print,
+  };
+
+  function createQueueElement(element: any, priority: number): IQueueElement {
+    return {
+      element,
+      priority,
+    };
+  }
+
+  function enqueue(element: any, priority: number): void {
+    const newElement: IQueueElement = createQueueElement(element, priority);
+    let added: boolean = false;
+
+    for (let index = 0; index < items.length; index++) {
+      const currentElement: IQueueElement = items[index];
+
+      if (newElement.priority < currentElement.priority) {
+        items.splice(index, 0, newElement);
+        added = true;
+        break;
+      }
+    }
+
+    if (!added) {
+      items.push(newElement);
+    }
+  }
+
+  function dequeue(): IQueueElement | undefined{
```

Comment:
This function is O(n) worst case. You should at least try to make one of the main operations faster